timesketch/frontend-ng/src/components/UploadForm.vue

```diff
@@ -0,0 +1,552 @@
+        )
+      }
+      let sources = []
+      let defaultValue = null
+      if (lastElementSelected === 'Create New Header') {
+        sources = null
+        do {
+          defaultValue = prompt('Insert the default value for this header')
+          if (defaultValue.includes(this.CSVDelimiter)) {
+            alert(`New header value cannot contain CSV separator (found ${this.CSVDelimiter})`)
+            defaultValue = null
+          }
+        } while (!defaultValue)
+      } else {
+        sources = this.mandatoryHeaders[i].columnsSelected
+      }
+
+      this.headersMapping = this.headersMapping.filter((mapping) => mapping['target'] !== target)
+      if (sources === null || sources.length > 0)
+        this.headersMapping.push({
+          target: target,
+          source: sources,
+          default_value: defaultValue,
+        })
+      this.validateFile()
+    },
+    clearFormData: function () {
+      this.fileMetaData = {}
+      this.form.name = ''
+      this.form.file = ''
+      this.fileName = ''
+      this.headersMapping = []
+      this.infoMessage = ''
+      this.headersString = ''
+      this.valuesString = []
+      this.uploadedFiles = []
+      this.title = 'Upload your Plaso/JSONL/CSV file'
+      this.error = []
+      this.mandatoryHeaders = [
+        { name: 'datetime', columnsSelected: [] },
+        { name: 'message', columnsSelected: [] },
+        { name: 'timestamp_desc', columnsSelected: [] },
+      ]
+    },
+    submitForm: function () {
+      if (!this.validateFile()) {
+        return
+      }
+
+      let formData = new FormData()
+      formData.append('file', this.form.file)
+      formData.append('name', this.form.name)
+      formData.append('provider', 'WebUpload')
+      formData.append('context', this.fileName)
+      formData.append('total_file_size', this.form.file.size)
+      formData.append('sketch_id', this.$store.state.sketch.id)
+      if (['csv', 'jsonl', 'json'].includes(this.extension)) {
+        let hMapping = JSON.stringify(this.headersMapping)
+        formData.append('headersMapping', hMapping)
+        formData.append('delimiter', this.CSVDelimiter)
+      }
+      let config = {
+        headers: {
+          'Content-Type': 'multipart/form-data',
+        },
+        onUploadProgress: function (progressEvent) {
+          this.percentCompleted = Math.round((progressEvent.loaded * 100) / progressEvent.total)
+        }.bind(this),
+      }
+      ApiClient.uploadTimeline(formData, config)
+        .then((response) => {
+          this.$store.dispatch('updateSketch', this.$store.state.sketch.id)
+          this.clearFormData()
+          this.percentCompleted = 0
+        })
+        .catch((e) => {})
+    },
+    validateFile: function () {
+      this.error = []
+      if (this.form.file.size <= 0) {
+        this.error.push('Please select a non empty file')
+      }
+      let allowedExtensions = ['csv', 'json', 'jsonl', 'plaso']
```

Comment:
Isn't json a bit misleading here? We treat any file with .json as JSONL, correct?
Maybe consider dropping json?

Reply:
That's a good point, @jaegeral and I discussed about it last week.

The answer for your first 2 questions is: YES. To this end, in the message for the client (when we ask for uploading a new file) we only specify JSONL/CSV/Plaso files. As you said, we treat (on purporse) any .JSON file as .JSONL because we assumed that there could be some users that upload a JSONL file with the JSON extension (e.g., OpenSOC challenge, all file are JSONL but with JSON extension).

If the file is actually a JSON then the parsing fails either client and server side. We could actually add a message saying the conversion failed, something like: "JSONL parsing failed: please add a JSONL/CSV/Plaso file instead."

What do you think?